### PR TITLE
migrate volume/csi/csi_mounter.go logs into structured logging

### PR DESCRIPTION
**Pull request by CKchen0726**
What type of PR is this?
/kind cleanup

What this PR does / why we need it:
part of kubernetes/enhancements#1602

Which issue(s) this PR fixes:
keps/sig-instrumentation/1602-structured-logging

Structured Logging migration instructions

Special notes for your reviewer:

Does this PR introduce a user-facing change?
```release-note
migrate volume/csi/csi_mounter.go logs into structured logging
```

### CODE REVIEW

File: pkg/volume/csi/csi_mounter.go [hunk 1]
```diff
@@ -106,7 +106,7 @@ func (c *csiMountMgr) SetUp(mounterArgs volume.MounterArgs) error {
 }
 
 func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error {
-	klog.V(4).Infof(log("Mounter.SetUpAt(%s)", dir))
+	klog.V(4).InfoS(log("Mounter.SetUpAt"), "dir", dir)
```
Comment (shubheksha) [inline pkg/volume/csi/csi_mounter.go:109]:
```suggestion
	klog.V(4).InfoS(log("Mounter.SetUpAt"), "path", dir)
```

Ref: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#name-arguments

File: pkg/volume/csi/csi_mounter.go [hunk 2]
```diff
@@ -208,7 +208,7 @@ func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error
 	if err := os.MkdirAll(parentDir, 0750); err != nil {
 		return errors.New(log("mounter.SetUpAt failed to create dir %#v:  %v", parentDir, err))
 	}
-	klog.V(4).Info(log("created target path successfully [%s]", parentDir))
+	klog.V(4).InfoS(log("Created target path successfully"), "parentDir", parentDir)
```
Comment (shubheksha) [inline pkg/volume/csi/csi_mounter.go:211]:
```suggestion
	klog.V(4).InfoS(log("Created target path successfully"), "path", parentDir)
```

Ref: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#name-arguments

File: pkg/volume/csi/csi_mounter.go [hunk 3]
```diff
@@ -254,15 +254,15 @@ func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error
 		// If operation finished with error then we can remove the mount directory.
 		if volumetypes.IsOperationFinishedError(err) {
 			if removeMountDirErr := removeMountDir(c.plugin, dir); removeMountDirErr != nil {
-				klog.Error(log("mounter.SetupAt failed to remove mount dir after a NodePublish() error [%s]: %v", dir, removeMountDirErr))
+				klog.ErrorS(removeMountDirErr, log("Mounter.SetUpAt failed to remove mount dir after a NodePublishVolume()"), "dir", dir)
```
Comment (shubheksha) [inline pkg/volume/csi/csi_mounter.go:257]:
```suggestion
				klog.ErrorS(removeMountDirErr, log("Mounter.SetUpAt failed to remove mount dir after a NodePublishVolume()"), "path", dir)
				
``````
Ref: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#name-arguments

File: pkg/volume/csi/csi_mounter.go [hunk 4]
```diff
@@ -275,10 +275,10 @@ func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error
 			//      cleaned up.
 			return volumetypes.NewUncertainProgressError(fmt.Sprintf("applyFSGroup failed for vol %s: %v", c.volumeID, err))
 		}
-		klog.V(4).Info(log("mounter.SetupAt fsGroup [%d] applied successfully to %s", *mounterArgs.FsGroup, c.volumeID))
+		klog.V(4).InfoS(log("Mounter.SetUpAt fsGroup applied successfully to volume"), "group", *mounterArgs.FsGroup, "volume", c.volumeID)
```
Comment (shubheksha) [inline pkg/volume/csi/csi_mounter.go:278]:
```suggestion
		klog.V(4).InfoS(log("Mounter.SetUpAt fsGroup applied successfully to volume"), "group", *mounterArgs.FsGroup, "volumeID", c.volumeID)
```

File: pkg/volume/csi/csi_mounter.go [hunk 5]
```diff
@@ -275,10 +275,10 @@ func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error
...
 	}
 
-	klog.V(4).Infof(log("mounter.SetUp successfully requested NodePublish [%s]", dir))
+	klog.V(4).InfoS(log("Mounter.SetUpAt successfully requested NodePublishVolume"), "dir", dir)
```
Comment (shubheksha) [inline pkg/volume/csi/csi_mounter.go:281]:
```suggestion
	klog.V(4).InfoS(log("Mounter.SetUpAt successfully requested NodePublishVolume"), "path", dir)
```

File: pkg/volume/csi/csi_mounter.go [hunk 6]
```diff
@@ -295,15 +295,15 @@ func (c *csiMountMgr) podAttributes() (map[string]string, error) {
 	csiDriver, err := c.plugin.csiDriverLister.Get(string(c.driverName))
 	if err != nil {
 		if apierrors.IsNotFound(err) {
-			klog.V(4).Infof(log("CSIDriver %q not found, not adding pod information", c.driverName))
+			klog.V(4).InfoS(log("CSIDriver not found, not adding pod information"), "driver", c.driverName)
```
Comment (shubheksha) [inline pkg/volume/csi/csi_mounter.go:298]:
```suggestion
			klog.V(4).InfoS(log("CSIDriver not found, not adding pod information"), "driverName", c.driverName)
```

File: pkg/volume/csi/csi_mounter.go [hunk 7]
```diff
@@ -295,15 +295,15 @@ func (c *csiMountMgr) podAttributes() (map[string]string, error) {
...
 			return nil, nil
 		}
 		return nil, err
 	}
 
 	// if PodInfoOnMount is not set or false we do not set pod attributes
 	if csiDriver.Spec.PodInfoOnMount == nil || *csiDriver.Spec.PodInfoOnMount == false {
-		klog.V(4).Infof(log("CSIDriver %q does not require pod information", c.driverName))
+		klog.V(4).InfoS(log("CSIDriver does not require pod information"), "driver", c.driverName)
```
Comment (shubheksha) [inline pkg/volume/csi/csi_mounter.go:306]:
```suggestion
		klog.V(4).InfoS(log("CSIDriver does not require pod information"), "driverName", c.driverName)
```

File: pkg/volume/csi/csi_mounter.go [hunk 8]
```diff
@@ -317,7 +317,7 @@ func (c *csiMountMgr) podAttributes() (map[string]string, error) {
 		attrs["csi.storage.k8s.io/ephemeral"] = strconv.FormatBool(c.volumeLifecycleMode == storage.VolumeLifecycleEphemeral)
 	}
 
-	klog.V(4).Infof(log("CSIDriver %q requires pod information", c.driverName))
+	klog.V(4).InfoS(log("CSIDriver requires pod information"), "driver", c.driverName)
```
Comment (shubheksha) [inline pkg/volume/csi/csi_mounter.go:320]:
```suggestion
	klog.V(4).InfoS(log("CSIDriver requires pod information"), "driverName", c.driverName)
```

File: pkg/volume/csi/csi_mounter.go [hunk 9]
```diff
@@ -329,7 +329,7 @@ func (c *csiMountMgr) podServiceAccountTokenAttrs() (map[string]string, error) {
 	csiDriver, err := c.plugin.csiDriverLister.Get(string(c.driverName))
 	if err != nil {
 		if apierrors.IsNotFound(err) {
-			klog.V(5).Infof(log("CSIDriver %q not found, not adding service account token information", c.driverName))
+			klog.V(5).InfoS(log("CSIDriver not found, not adding service account token information"), "driver", c.driverName)
```
Comment (shubheksha) [inline pkg/volume/csi/csi_mounter.go:332]:
```suggestion
			klog.V(5).InfoS(log("CSIDriver not found, not adding service account token information"), "driverName", c.driverName)
```

File: pkg/volume/csi/csi_mounter.go [hunk 10]
```diff
@@ -365,7 +365,7 @@ func (c *csiMountMgr) podServiceAccountTokenAttrs() (map[string]string, error) {
 		outputs[audience] = tr.Status
 	}
 
-	klog.V(4).Infof(log("Fetched service account token attrs for CSIDriver %q", c.driverName))
+	klog.V(4).InfoS(log("Fetched service account token attrs for CSIDriver"), "driver", c.driverName)
```
Comment (shubheksha) [inline pkg/volume/csi/csi_mounter.go:368]:
```suggestion
	klog.V(4).InfoS(log("Fetched service account token attrs for CSIDriver"), "driverName", c.driverName)
```

File: pkg/volume/csi/csi_mounter.go [hunk 11]
```diff
@@ -387,27 +387,27 @@ func (c *csiMountMgr) TearDown() error {
 	return c.TearDownAt(c.GetPath())
 }
 func (c *csiMountMgr) TearDownAt(dir string) error {
-	klog.V(4).Infof(log("Unmounter.TearDown(%s)", dir))
+	klog.V(4).InfoS(log("Unmounter.TearDownAt"), "dir", dir)
```
Comment (shubheksha) [inline pkg/volume/csi/csi_mounter.go:390]:
```suggestion
	klog.V(4).InfoS(log("Unmounter.TearDownAt"), "path", dir)
```

File: pkg/volume/csi/csi_mounter.go [hunk 12]
```diff
@@ -387,27 +387,27 @@ func (c *csiMountMgr) TearDown() error {
...
+		return errors.New(log("unmounter.TearDownAt failed: %v", err))
 	}
 
 	// clean mount point dir
 	if err := removeMountDir(c.plugin, dir); err != nil {
-		return errors.New(log("mounter.TearDownAt failed to clean mount dir [%s]: %v", dir, err))
+		return errors.New(log("unmounter.TearDownAt failed to clean mount dir [%s]: %v", dir, err))
 	}
-	klog.V(4).Infof(log("mounter.TearDownAt successfully unmounted dir [%s]", dir))
+	klog.V(4).InfoS(log("Unmounter.TearDownAt successfully unmounted dir"), "dir", dir)
```
Comment (shubheksha) [inline pkg/volume/csi/csi_mounter.go:410]:
```suggestion
	klog.V(4).InfoS(log("Unmounter.TearDownAt successfully unmounted dir"), "path", dir)
```

File: pkg/volume/csi/csi_mounter.go [hunk 13]
```diff
@@ -443,7 +443,7 @@ func isDirMounted(plug *csiPlugin, dir string) (bool, error) {
 	mounter := plug.host.GetMounter(plug.GetPluginName())
 	notMnt, err := mounter.IsLikelyNotMountPoint(dir)
 	if err != nil && !os.IsNotExist(err) {
-		klog.Error(log("isDirMounted IsLikelyNotMountPoint test failed for dir [%v]", dir))
+		klog.ErrorS(err, log("IsLikelyNotMountPoint check failed for dir"), "dir", dir)
```
Comment (shubheksha) [inline pkg/volume/csi/csi_mounter.go:446]:
```suggestion
		klog.ErrorS(err, log("IsLikelyNotMountPoint check failed for dir"), "path", dir)
```

File: pkg/volume/csi/csi_mounter.go [hunk 14]
```diff
@@ -456,26 +456,26 @@ func isCorruptedDir(dir string) bool {
...
-		klog.V(4).Info(log("dir not mounted, deleting it [%s]", mountPath))
+		klog.V(4).InfoS(log("Dir not mounted, deleting the mount path"), "path", mountPath)
 		if err := os.Remove(mountPath); err != nil && !os.IsNotExist(err) {
 			return errors.New(log("failed to remove dir [%s]: %v", mountPath, err))
 		}
 		// remove volume data file as well
 		volPath := filepath.Dir(mountPath)
 		dataFile := filepath.Join(volPath, volDataFileName)
-		klog.V(4).Info(log("also deleting volume info data file [%s]", dataFile))
+		klog.V(4).InfoS(log("Remove volume info data file"), "dataFile", dataFile)
```
Comment (shubheksha) [inline pkg/volume/csi/csi_mounter.go:473]:
```suggestion
		klog.V(4).InfoS(log("Remove volume info data file"), "dataFilePath", dataFile)
```

File: pkg/volume/csi/csi_mounter.go [hunk 15]
```diff
@@ -456,26 +456,26 @@ func isCorruptedDir(dir string) bool {
...
 		if err := os.Remove(dataFile); err != nil && !os.IsNotExist(err) {
 			return errors.New(log("failed to delete volume data file [%s]: %v", dataFile, err))
 		}
 		// remove volume path
-		klog.V(4).Info(log("deleting volume path [%s]", volPath))
+		klog.V(4).InfoS(log("Remove volume path"), "volPath", volPath)
```
Comment (shubheksha) [inline pkg/volume/csi/csi_mounter.go:478]:
```suggestion
		klog.V(4).InfoS(log("Remove volume path"), "volumePath", volPath)
```

**Review by shubheksha — CHANGES_REQUESTED** [review]
Left a few comments around naming arguments, looks good otherwise 👍

File: pkg/volume/csi/csi_mounter.go [hunk 16]
```diff
@@ -254,15 +254,15 @@ func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error
...
+				klog.ErrorS(removeMountDirErr, log("Mounter.SetUpAt failed to remove mount dir after a NodePublishVolume()"), "dir", dir)
 			}
 		}
 		return err
 	}
 
 	c.supportsSELinux, err = c.kubeVolHost.GetHostUtil().GetSELinuxSupport(dir)
 	if err != nil {
-		klog.V(2).Info(log("error checking for SELinux support: %s", err))
+		klog.V(2).ErrorS(err, log("Failed to check SELinux support"))
```
Comment (yangjunmyfm192085) [inline pkg/volume/csi/csi_mounter.go:265]: Do we need to modify Info to ErrorS here?
May be InfoS is ok?

**Comment by shubheksha** [comment]
@CKchen0726 are you still planning to work on this?

File: pkg/volume/csi/csi_mounter.go (hunk 9 above)
Comment (CKchen0726) [reply pkg/volume/csi/csi_mounter.go:294]: Thanks, I will update it later.

**Comment by shubheksha** [comment]
@CKchen0726 are you still planning to work on this?

**Pull request closed without merging**